src/coreComponents/common/LifoStorageCommon.hpp

```diff
@@ -0,0 +1,295 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+#ifndef LIFOSTORAGECOMMON_HPP
+#define LIFOSTORAGECOMMON_HPP
+
+#include <deque>
+#include <future>
+#include <mutex>
+#include <condition_variable>
+#include <camp/camp.hpp>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <algorithm>
+
+#ifdef LIFO_DISABLE_CALIPER
+#define LIFO_MARK_FUNCTION
+#define LIFO_MARK_SCOPE( a )
+#define LIFO_LOG_RANK( a ) std::cerr << a << std::endl;
+#else
+#define LIFO_MARK_FUNCTION GEOSX_MARK_FUNCTION
+#define LIFO_MARK_SCOPE( a ) GEOSX_MARK_SCOPE( a )
+#define LIFO_LOG_RANK( a ) GEOSX_LOG_RANK( a )
+#endif
+
+#include "common/GEOS_RAJA_Interface.hpp"
+#include "common/TimingMacros.hpp"
+#include "common/FixedSizeDequeWithMutexes.hpp"
+#include "common/TwoMutexesLock.hpp"
+
+
+namespace geosx
+{
+/**
+ * This class is used to store in a LIFO way buffers, first on device, then on host, then on disk.
+ */
+template< typename T, typename INDEX_TYPE >
+class LifoStorageCommon
+{
+protected:
```

Comment:
Please put `protected` and `private` components at the end of the class.

Reply:
ok